lane.py

```diff
@@ -1921,6 +1930,38 @@ def _gather_genre_ids(self):
             )
         return genre_ids
 
+    @property
+    def customlist_ids(self):
+        """Find the database ID of every CustomList such that a Work filed
+        in that List should be in this Lane.
+
+        :return: A list of CustomList IDs, possibly empty.
+        """
+        if not hasattr(self, '_customlist_ids'):
+            self._customlist_ids = self._gather_customlist_ids()
+        return self._customlist_ids
+
+    def _gather_customlist_ids(self):
+        """Method that does the work of `customlist_ids`."""
+        if self.list_datasource:
+            # Find the ID of every CustomList from a certain
+            # DataSource.
+            _db = Session.object_session(self)
+            query = select(
+                [CustomList.id],
+                CustomList.data_source_id==self.list_datasource.id
+            )
+            ids = [x[0] for x in _db.execute(query)]
+        else:
+            # Find the IDs of some specific CustomLists.
+            ids = [x.id for x in self.customlists]
+        if len(ids) == 0:
+            # Return None to make it clear that we mean "there is no
+            # custom list restriction" rather than "exclude
+            # everything".
+            return None
```

Comment:
If the lane is based on a list data source and there are no lists with that data source yet, I think this should return an empty list rather than None.